server/src/main/java/com/cloud/storage/StoragePoolAutomationImpl.java

```diff
@@ -318,9 +323,12 @@ public boolean cancelMaintain(DataStore store) {
         if (hosts == null || hosts.size() == 0) {
             return true;
         }
+
+        Map<String, String> details = new HashMap<>();
```

Comment:
can create this map in the method itself and return (noticed map is created before _addStoragePoolNFSMountOptsToDetailsMap()_ is called )

Reply:
I kept it outside thinking that other kind of details can be added to `details` map later

For ex. we can have something like this in future
```
Map<String, String> details = new HashMap<>();
storageManager.addStoragePoolNFSMountOptsToDetailsMap(pool, details);
storageManager.addStoragePoolFutureFieldToDetailsMap(pool, details);
```
I can move it to the method if you think this is overkill.

Reply:
if details passed is null, create it and add detail. otherwise, append to it. (method would return updated details map)